atomic_sequence.go

```diff
@@ -0,0 +1,281 @@
+package sequence
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync/atomic"
+)
+
+// NewAtomic creates a NewSequence that implements the Incrementer interface.
+// It is safe for concurrent use.
+func NewAtomic(params ...uint64) (*AtomicSequence, error) {
+	seq := new(AtomicSequence)
+	err := seq.Init(params...)
+	return seq, err
+}
+
+// AtomicSequence is a basic Sequence that uses atomic instructions in Sequence methods.
+// Although implementation is very close, it is safe for concurrent use.
+type AtomicSequence Sequence
```

Comment:
Very cool - thanks. Out of curiosity if I had implemented concurrent accesses I might have selected a `sync.RWMutex` -- do you have a feeling for the performance of `atomic` vs `sync`? I get the impression that `atomic` [is faster](https://blog.dubbelboer.com/2015/08/23/rwmutex-vs-atomicvalue-vs-unsafepointer.html) but here each method has multiple atomic accesses, rather than one synchronization barrier.

Reply:
Hmm I have no idea on this case as I never benchmarked one over the other. I agree that as there is usually 3-4 calls it might be slower on this path too.

I think lock contention would be worse though, but I am really not sure...

Reply:
Shouldn't be too hard to test - I'll check it out and let you know, though I do think `atomic` will be faster.